lib/datastore/query.js

```diff
@@ -278,9 +269,22 @@ Query.prototype.limit = function(n) {
  * var offsetQuery = companyQuery.offset(100);
  */
 Query.prototype.offset = function(n) {
-  var query = extend(true, new Query(), this);
-  query.offsetVal = n;
-  return query;
+  this.offsetVal = n;
+  return this;
+};
+
+/**
+ * Creates a clone of a query. This is useful if you want to work with a Query
+ * in an immutable fashion.
+ *
+ * @return {module:datastore/query}
+ *
+ * @example
+ * var query = dataset.createQuery('Lion');
+ * var limitedQuery = query.clone().limit(10);
+ */
+Query.prototype.clone = function() {
```

Comment:
I prefer getting rid of things that we've found to not really be necessary instead of dragging support for them along. Before we lock onto a major version, it's a great time.